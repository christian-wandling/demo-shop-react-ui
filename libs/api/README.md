# api

A comprehensive library for consuming the demo-shop api. Supports accessing a product catalog, user management,
shopping cart operations, and order processing.
