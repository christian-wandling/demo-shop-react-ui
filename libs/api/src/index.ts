export * from './lib/openapi';
export * from './lib/ApiContext';
export * from './lib/services/apiService';
export * from './lib/ApiProvider';
export * from './lib/services/apiService';
export * from './lib/middleware/authMiddleware';
