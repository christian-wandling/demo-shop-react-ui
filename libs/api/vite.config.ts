import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';

export default defineConfig(() => ({
  root: __dirname,
  cacheDir: '../../node_modules/.vite/libs/api',
  plugins: [react()],
  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },
  test: {
    watch: false,
    globals: true,
    environment: 'jsdom',
    reporters: ['default'],
    passWithNoTests: true,
    coverage: {
      reportsDirectory: './test-output/vitest/coverage',
      enabled: false,
    },
  },
}));
