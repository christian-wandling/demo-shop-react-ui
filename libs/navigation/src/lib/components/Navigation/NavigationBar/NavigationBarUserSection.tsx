import { UserNavigationProps } from '../../../models/userNavigationProps';

export const NavigationBarUserSection = ({ user, onLogin, onLogout, onRegister }: UserNavigationProps) => {
  if (user) {
    return (
      <div className="hidden lg:flex lg:flex-1 lg:items-center lg:justify-end lg:space-x-6">
        <a className="text-sm font-medium text-gray-700  cursor-pointer">
          {user.firstname}&nbsp;{user.lastname}
        </a>
        <span className="h-6 w-px bg-gray-200" aria-hidden="true"></span>
        <a onClick={onLogout} className="text-sm font-medium text-gray-700 hover:text-indigo-500 cursor-pointer">
          Sign out
        </a>
      </div>
    );
  }

  return (
    <div className="hidden lg:flex lg:flex-1 lg:items-center lg:justify-end lg:space-x-6">
      <a onClick={onLogin} className="text-sm font-medium text-gray-700 hover:text-indigo-500 cursor-pointer">
        Sign in
      </a>
      <span className="h-6 w-px bg-gray-200" aria-hidden="true"></span>
      <a onClick={onRegister} className="text-sm font-medium text-gray-700 hover:text-indigo-500 cursor-pointer">
        Register
      </a>
    </div>
  );
};
