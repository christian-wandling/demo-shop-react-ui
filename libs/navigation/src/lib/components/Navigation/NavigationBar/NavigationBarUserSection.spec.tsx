import { describe, expect, it, vi } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import { NavigationBarUserSection } from './NavigationBarUserSection';
import { UserResponse } from '@demo-shop-react-ui/api';

describe('NavigationBarUserSection', () => {
  const mockUser: UserResponse = {
    email: 'email',
    id: 0,
    phone: null,
    firstname: 'John',
    lastname: 'Doe',
  };

  const mockHandlers = {
    onLogin: vi.fn(),
    onLogout: vi.fn(),
    onRegister: vi.fn(),
  };

  it('renders login and register buttons when user is not logged in', () => {
    render(
      <NavigationBarUserSection
        user={undefined}
        onLogin={mockHandlers.onLogin}
        onLogout={mockHandlers.onLogout}
        onRegister={mockHandlers.onRegister}
      />
    );

    expect(screen.getByText('Sign in')).toBeTruthy();
    expect(screen.getByText('Register')).toBeTruthy();
    expect(screen.queryByText('Sign out')).not.toBeTruthy();
  });

  it('renders user name and logout button when user is logged in', () => {
    render(
      <NavigationBarUserSection
        user={mockUser}
        onLogin={mockHandlers.onLogin}
        onLogout={mockHandlers.onLogout}
        onRegister={mockHandlers.onRegister}
      />
    );

    expect(screen.getByText('John Doe')).toBeTruthy();
    expect(screen.getByText('Sign out')).toBeTruthy();
    expect(screen.queryByText('Sign in')).not.toBeTruthy();
    expect(screen.queryByText('Register')).not.toBeTruthy();
  });

  it('calls onLogin when Sign in is clicked', () => {
    render(
      <NavigationBarUserSection
        user={undefined}
        onLogin={mockHandlers.onLogin}
        onLogout={mockHandlers.onLogout}
        onRegister={mockHandlers.onRegister}
      />
    );

    fireEvent.click(screen.getByText('Sign in'));
    expect(mockHandlers.onLogin).toHaveBeenCalledTimes(1);
  });

  it('calls onRegister when Register is clicked', () => {
    render(
      <NavigationBarUserSection
        user={undefined}
        onLogin={mockHandlers.onLogin}
        onLogout={mockHandlers.onLogout}
        onRegister={mockHandlers.onRegister}
      />
    );

    fireEvent.click(screen.getByText('Register'));
    expect(mockHandlers.onRegister).toHaveBeenCalledTimes(1);
  });

  it('calls onLogout when Sign out is clicked', () => {
    render(
      <NavigationBarUserSection
        user={mockUser}
        onLogin={mockHandlers.onLogin}
        onLogout={mockHandlers.onLogout}
        onRegister={mockHandlers.onRegister}
      />
    );

    fireEvent.click(screen.getByText('Sign out'));
    expect(mockHandlers.onLogout).toHaveBeenCalledTimes(1);
  });

  it('displays the correct user name format', () => {
    render(
      <NavigationBarUserSection
        user={{ ...mockUser, firstname: 'Jane', lastname: 'Smith' }}
        onLogin={mockHandlers.onLogin}
        onLogout={mockHandlers.onLogout}
        onRegister={mockHandlers.onRegister}
      />
    );

    expect(screen.getByText('Jane Smith')).toBeTruthy();
  });

  describe('Snapshots', () => {
    it('matches the snapshot logged out state', () => {
      const { baseElement } = render(
        <NavigationBarUserSection
          user={undefined}
          onLogin={mockHandlers.onLogin}
          onLogout={mockHandlers.onLogout}
          onRegister={mockHandlers.onRegister}
        />
      );

      expect(baseElement.firstChild).toMatchSnapshot();
    });

    it('matches the snapshot logged in state', () => {
      const { baseElement } = render(
        <NavigationBarUserSection
          user={mockUser}
          onLogin={mockHandlers.onLogin}
          onLogout={mockHandlers.onLogout}
          onRegister={mockHandlers.onRegister}
        />
      );

      expect(baseElement.firstChild).toMatchSnapshot();
    });
  });
});
