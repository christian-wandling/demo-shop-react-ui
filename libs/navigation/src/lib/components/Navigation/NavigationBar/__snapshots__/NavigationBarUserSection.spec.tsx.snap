// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationBarUserSection > Snapshots > matches the snapshot logged in state 1`] = `
<div>
  <div
    class="hidden lg:flex lg:flex-1 lg:items-center lg:justify-end lg:space-x-6"
  >
    <a
      class="text-sm font-medium text-gray-700  cursor-pointer"
    >
      John
      Â 
      Doe
    </a>
    <span
      aria-hidden="true"
      class="h-6 w-px bg-gray-200"
    />
    <a
      class="text-sm font-medium text-gray-700 hover:text-indigo-500 cursor-pointer"
    >
      Sign out
    </a>
  </div>
</div>
`;

exports[`NavigationBarUserSection > Snapshots > matches the snapshot logged out state 1`] = `
<div>
  <div
    class="hidden lg:flex lg:flex-1 lg:items-center lg:justify-end lg:space-x-6"
  >
    <a
      class="text-sm font-medium text-gray-700 hover:text-indigo-500 cursor-pointer"
    >
      Sign in
    </a>
    <span
      aria-hidden="true"
      class="h-6 w-px bg-gray-200"
    />
    <a
      class="text-sm font-medium text-gray-700 hover:text-indigo-500 cursor-pointer"
    >
      Register
    </a>
  </div>
</div>
`;
